base/array.jl

```diff
@@ -984,7 +984,7 @@ push!(a::AbstractVector, iter...) = append!(a, iter)
 function _append!(a, ::Union{HasLength,HasShape}, iter)
     n = length(a)
     i = lastindex(a)
-    resize!(a, n+length(iter))
+    resize!(a, n+length(iter)::Int)
```

Comment:
Also this and the next one; arbitrary iterators might not have `Int` length.

Reply:
You don't mind converting to Int and then asserting?